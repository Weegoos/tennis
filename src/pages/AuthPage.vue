<template>
  <div
    class="fixed-center"
    style="display: flex; height: 100vh; width: 100vw"
    @keydown="handleKey"
  >
    <!-- Левая половина с видео -->
    <div
      style="flex: 1; position: relative; overflow: hidden"
      v-if="$q.screen.width > mobileWidth"
    >
      <video
        src="../assets/background.mp4"
        autoplay
        muted
        loop
        playsinline
        style="width: 100%; height: 100%; object-fit: cover"
      ></video>
    </div>

    <!-- Правая половина с текстом -->
    <div class="content">
      <q-card class="my-card" style="width: 100%">
        <q-card-section align="center">
          <p
            class="text-bold"
            :class="$q.screen.width > mobileWidth ? 'text-h4' : 'text-h6'"
            data-testid="authorizationContent"
          >
            Authorization
          </p>
        </q-card-section>

        <q-card-section>
          <div
            class="q-gutter-sm"
            :class="$q.screen.width > mobileWidth ? 'row' : 'column'"
          >
            <div class="col">
              <q-input
                v-model="email"
                data-testid="emailInput"
                type="email"
                label="Enter your email address"
                hint="For example: example@gmail.com"
              />
            </div>
            <div class="col">
              <q-input
                v-model="password"
                data-testid="passwordInput"
                :type="isPwd ? 'password' : 'text'"
                hint="At least 8 characters"
                label="Enter the password"
              >
                <template v-slot:append>
                  <q-icon
                    :name="isPwd ? 'visibility_off' : 'visibility'"
                    class="cursor-pointer"
                    @click="isPwd = !isPwd"
                  />
                </template>
              </q-input>
            </div>
          </div>
        </q-card-section>

        <q-card-actions vertical>
          <q-btn
            color="positive"
            no-caps
            label="Enter"
            data-testid="authorizationButton"
            rounded
            @click="authorization"
          />
          <q-btn
            color="primary"
            no-caps
            flat
            label="Don't have an account?"
            @click="pushToRegistration"
          />
        </q-card-actions>
      </q-card>
    </div>
  </div>
</template>

<script setup>
import axios from "axios";
import { Cookies, QSpinnerGears, useQuasar } from "quasar";
import { ref } from "vue";
import { useRouter } from "vue-router";
import { getCurrentInstance } from "vue";
import { useNotifyStore } from "src/stores/notify-store";
import { useApiStore } from "src/stores/api-store";

// global variables
const { proxy } = getCurrentInstance();
const serverURL = proxy.$serverURL;
const mobileWidth = proxy.$mobileWidth;
const $q = useQuasar();
const notifyStore = useNotifyStore();
const router = useRouter();

// variables
const isPwd = ref(true);

// function
const email = ref("");
const password = ref("");

const authorization = async () => {
  notifyStore.loading($q, "Подождите...", QSpinnerGears);
  try {
    const data = {
      email: email.value,
      password: password.value,
    };
    const response = await axios.post(`${serverURL}auth/signin`, data, {
      headers: {
        "Content-Type": "application/json",
      },
      withCredentials: true,
    });

    $q.loading.hide();
    notifyStore.nofifySuccess($q, `Авторизация прошла успешно!`);

    Cookies.set("accessToken", response.data.accessToken);
    Cookies.set("refreshToken", response.data.refreshToken);

    console.log("Токен после установки:", Cookies.get("accessToken"));

    setTimeout(() => {
      router.push("/");
    }, 50);
  } catch (error) {
    $q.loading.hide();
    notifyStore.notifyError($q, "Ошибка при авторизации. Попробуйте снова.");
  }
};

const pushToRegistration = () => {
  router.push("/registration");
};

const handleKey = (e) => {
  if (e.key === "Enter") {
    authorization();
  }
};
</script>

<style scoped>
.carouselClass {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 80%;
  justify-content: center;
  align-items: center;
}

.transparent-carousel {
  background-color: rgba(0, 0, 0, 0.5);
  box-shadow: none;
}

.content {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 20px;
  background-color: rgba(255, 255, 255, 0.8);
}
</style>
