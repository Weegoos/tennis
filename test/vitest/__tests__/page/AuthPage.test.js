import { installQuasarPlugin } from "@quasar/quasar-app-extension-testing-unit-vitest";
import { mount } from "@vue/test-utils";
import { Quasar } from "quasar";
import AuthPage from "src/pages/AuthPage.vue";
import { describe, expect, it } from "vitest";
installQuasarPlugin();

describe("tests for AuthPage.vue", () => {
  const wrapper = mount(AuthPage, {
    global: {
      plugins: [Quasar],
    },
  });
  it("should find authorizationContent data-testid", () => {
    const authorizationContent = wrapper.find(
      '[data-testid="authorizationContent"]'
    );
    expect(authorizationContent.exists()).toBe(true);
    expect(authorizationContent.text()).toContain("Authorization");
    expect(authorizationContent).toMatchSnapshot();
  });
  it("should find emailInput data-testid", () => {
    const emailInput = wrapper.find('[data-testid="emailInput"]');
    expect(emailInput.exists()).toBe(true);

    emailInput.setValue("Example@gmail.com");
    expect(emailInput.element.value).toBe("Example@gmail.com");
  });

  it("should find passwordInput data-testid", () => {
    const passwordInput = wrapper.find('[data-testid="passwordInput"]');
    expect(passwordInput.exists()).toBe(true);

    passwordInput.setValue("examplePassword");
    expect(passwordInput.element.value).toBe("examplePassword");
  });
});
