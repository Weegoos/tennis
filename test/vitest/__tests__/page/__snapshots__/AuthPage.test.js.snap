// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for AuthPage.vue > should find authorizationContent data-testid 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <p
    aling="center"
    class="text-bold text-center text-h6 text-white"
    data-testid="authorizationContent"
    data-v-c025950a=""
  >
    Authorization
  </p>,
}
`;
